fe/fe-core/src/main/java/com/starrocks/common/util/concurrent/lock/Locker.java

```diff
@@ -242,18 +242,20 @@ public boolean isWriteLockHeldByCurrentThread(Database database) {
      */
     public void lockTablesWithIntensiveDbLock(Database database, List<Long> tableList, LockType lockType) {
         assert lockType == LockType.WRITE || lockType == LockType.READ;
+        List<Long> tableListClone = new ArrayList<>(tableList);
+
         if (Config.use_lock_manager) {
-            assert !tableList.isEmpty();
+            assert !tableListClone.isEmpty();
```

Comment:
what is the difference between assert and preconditions?

Reply:
I think there is no difference, but we can agree on a code specification. Do we need to use preconditions as precondition judgment to avoid bugs in subsequent code development?